pkg/kubelet/kubelet_getters.go

```diff
@@ -331,6 +331,20 @@ func (kl *Kubelet) getMountedVolumePathListFromDisk(podUID types.UID) ([]string,
 		if !isNotMount {
 			mountedVolumes = append(mountedVolumes, volumePath)
 		}
+
+		// This is a temporary solution for checking CSI mount path due to the extra "/mount".
+		// For CSI volumes, csiMountVolumePath is the right path to check.
+		// For volumes that are not CSI type, if this mount check returns true,
+		// it is ok to added to mountedVolumes because it also means that there are volumes exist and
+		// it is not safe to delete pod yet.
+		// We might remove this checking mount point as a condition before deleting pods since it is
+		// not reliable and the condition might be different for different types of volumes. But it first requires
+		// a reliable way to clean up volume dir. See discussion in issue #74650
+		csiMountVolumePath := filepath.Join(volumePath, "/mount")
```

Comment:
I think we should do this only if the volumePath is for a csi volume. We should avoid peeking into regular volume data as much as possible. 

Also add a comment that this check is still problematic for csi volumes that either: 1) don't mount or 2) bind mount in the rootfs